tests/test_api.py

```diff
@@ -446,3 +447,52 @@ def test_pop():
         respx.get("https://foo.bar/", alias="foobar")
         request_pattern = respx.pop("foobar")
         assert request_pattern.url == "https://foo.bar/"
+
+
+@respx.mock
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "url,params,call_url,call_params",
+    [
+        ("https://foo/", "foo=bar", "https://foo/", "foo=bar"),
+        ("https://foo/", b"foo=bar", "https://foo/", b"foo=bar"),
+        ("https://foo/", [("foo", "bar")], "https://foo/", [("foo", "bar")]),
+        ("https://foo/", {"foo": "bar"}, "https://foo/", {"foo": "bar"}),
+        ("https://foo?foo=bar", "baz=qux", "https://foo?foo=bar", "baz=qux"),
+        ("https://foo?foo=bar", "baz=qux", "https://foo?foo=bar&baz=qux", None),
+        (re.compile(r"https://foo/(\w+)/"), "foo=bar", "https://foo/bar/", "foo=bar"),
+        (httpx.URL("https://foo/"), "foo=bar", "https://foo/", "foo=bar"),
+        (
+            httpx.URL("https://foo?foo=bar"),
+            "baz=qux",
+            "https://foo?foo=bar&baz=qux",
+            None,
+        ),
+    ],
+)
+async def test_params_match(client, url, params, call_url, call_params):
+    respx.get(url, params=params, content="spam spam")
+    response = await client.get(call_url, params=call_params)
+    assert response.text == "spam spam"
+
+
+# @pytest.mark.asyncio
+# @pytest.mark.parametrize(
+#    "base,url",
+#    [
+#        (None, "https://foo.bar/baz"),
+#        ("", "https://foo.bar/baz"),
+#        ("https://foo.bar", "baz"),
+#        ("https://foo.bar/", "baz"),
+#        ("https://foo.bar/", "/baz"),
+#        ("https://foo.bar/", "/baz/"),
+#        ("https://foo.bar/baz", None),
+#        ("https://foo.bar/baz/", None),
+#        ("https://foo.bar/", re.compile(r"/(\w+)")),
+#    ],
+# )
+# async def test_build_url_base(client, base, url):
+#    with respx.mock(base_url=base):
+#        respx.get(url, content="spam spam")
```

Comment:
The docs are not clear enough on this, but when using the context manager with settings, i.e. parentheses, you should mock responses using the returned mock instance.

```suggestion
#    with respx.mock(base_url=base) as respx_mock:
#        respx_mock.get(url, content="spam spam")
```

Maybe that helps?

Although, it should not fail IMO, only ignore the settings for the mocked response. Let's remember this and try to reproduce the potential leakage in a separate PR.